src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpRequestStream.cs

```diff
@@ -122,11 +144,11 @@ public override int Read(byte[] buffer, int offset, int count)
             throw new NotSupportedException();
         }
 
-        internal void EndUpload()
+        internal async Task EndUploadAsync(CancellationToken token)
```

Comment:
It totally breaks object encapsulation to inline this code into the caller (WinHttpHandler). The implementation detail of the WinHttpRequestStream.EndUpload() method is to write the final zero chunk to the network stream (assuming it's chunked).  But to inline that code into the WinHttpHandler call would expose the internals of the WinHttpRequestStream implementation.

It still has to be a separate await-able operation since it is a separate network write operation.

Would marking the EndUpload method as this make it better?

```c#
        [MethodImpl(MethodImplOptions.AggressiveInlining)]
```